server/src/test/java/org/opensearch/cluster/metadata/MetadataIndexTemplateServiceTests.java

```diff
@@ -2155,31 +2154,7 @@ public void onFailure(Exception e) {
     }
 
     private MetadataIndexTemplateService getMetadataIndexTemplateService() {
-        IndicesService indicesService = getInstanceFromNode(IndicesService.class);
-        ClusterService clusterService = getInstanceFromNode(ClusterService.class);
-        MetadataCreateIndexService createIndexService = new MetadataCreateIndexService(
-            Settings.EMPTY,
-            clusterService,
-            indicesService,
-            null,
-            null,
-            createTestShardLimitService(randomIntBetween(1, 1000)),
-            new Environment(builder().put(Environment.PATH_HOME_SETTING.getKey(), createTempDir().toString()).build(), null),
-            IndexScopedSettings.DEFAULT_SCOPED_SETTINGS,
-            null,
-            xContentRegistry(),
-            new SystemIndices(Collections.emptyMap()),
-            true,
-            new AwarenessReplicaBalance(Settings.EMPTY, clusterService.getClusterSettings())
-        );
-        return new MetadataIndexTemplateService(
-            clusterService,
-            createIndexService,
-            new AliasValidator(),
-            indicesService,
-            new IndexScopedSettings(Settings.EMPTY, IndexScopedSettings.BUILT_IN_INDEX_SETTINGS),
-            xContentRegistry()
-        );
```

Comment:
why did we remove this?

Reply:
This method was creating `MetadataCreateIndexService` and `MetadataIndexTemplateService` for each test case. During each time it was registering tasks again and tests were getting failed.

Since this test case was extending `OpenSearchSingleNodeTestCase` we can get the instance of the any service by injector available on super class. 

That's why we removed it and getting service from injector via `getInstanceFromNode(class<>)`. We haven't changed anything in testcases but just this utility method and verified that all testcases are passing after making this change.